diff -Nru ACE_wrappers.orig/TAO/orbsvcs/orbsvcs/Makefile.am ACE_wrappers/TAO/orbsvcs/orbsvcs/Makefile.am
--- ACE_wrappers.orig/TAO/orbsvcs/orbsvcs/Makefile.am	2007-01-04 14:06:26.000000000 +0000
+++ ACE_wrappers/TAO/orbsvcs/orbsvcs/Makefile.am	2007-02-07 12:46:11.000000000 +0000
@@ -76,7 +76,7 @@
   Naming/Naming_Client.cpp
 
 libTAO_CosNaming_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosNaming_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_AnyTypeCode.la \
@@ -181,7 +181,7 @@
   CosPropertyServiceC.cpp
 
 libTAO_CosProperty_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosProperty_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_AnyTypeCode.la \
@@ -230,7 +230,7 @@
   CosPropertyServiceS.cpp
 
 libTAO_CosProperty_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosProperty_Skel_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PortableServer.la \
@@ -280,7 +280,7 @@
   Property/CosPropertyService_i.cpp
 
 libTAO_CosProperty_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosProperty_Serv_la_LIBADD = \
   libTAO_CosProperty_Skel.la \
@@ -424,7 +424,7 @@
   sfpS.cpp
 
 libTAO_AV_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_AV_la_LIBADD = \
   libTAO_CosNaming.la \
@@ -547,7 +547,7 @@
   CosConcurrencyControlC.cpp
 
 libTAO_CosConcurrency_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosConcurrency_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_AnyTypeCode.la \
@@ -593,7 +593,7 @@
   CosConcurrencyControlS.cpp
 
 libTAO_CosConcurrency_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosConcurrency_Skel_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PortableServer.la \
@@ -645,7 +645,7 @@
   Concurrency/Concurrency_Utils.cpp
 
 libTAO_CosConcurrency_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosConcurrency_Serv_la_LIBADD = \
   libTAO_CosConcurrency_Skel.la \
@@ -821,7 +821,7 @@
   CosTypedEventCommC.cpp
 
 libTAO_CosEvent_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosEvent_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_AnyTypeCode.la \
@@ -926,7 +926,7 @@
   Time_Utilities.cpp
 
 libTAO_Svc_Utils_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Svc_Utils_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PortableServer.la \
@@ -990,7 +990,7 @@
   CosTypedEventCommS.cpp
 
 libTAO_CosEvent_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosEvent_Skel_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PortableServer.la \
@@ -1079,7 +1079,7 @@
   CosEvent/CEC_TypedSupplierAdmin.cpp
 
 libTAO_CosEvent_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosEvent_Serv_la_LIBADD = \
   libTAO_Svc_Utils.la \
@@ -1229,7 +1229,7 @@
   LifeCycleServiceS.cpp
 
 libTAO_CosLifeCycle_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosLifeCycle_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PortableServer.la \
@@ -1392,7 +1392,7 @@
   miopS.cpp
 
 libTAO_PortableGroup_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_PortableGroup_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_Messaging.la \
@@ -1570,7 +1570,7 @@
   LoadBalancing/LB_ServerRequestInterceptor.cpp
 
 libTAO_CosLoadBalancing_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosLoadBalancing_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_IORInterceptor.la \
@@ -1662,7 +1662,7 @@
   CosNamingS.cpp
 
 libTAO_CosNaming_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosNaming_Skel_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PortableServer.la \
@@ -1724,7 +1724,7 @@
   Naming/Transient_Naming_Context.cpp
 
 libTAO_CosNaming_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosNaming_Serv_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_IORTable.la \
@@ -1953,7 +1953,7 @@
   NotifyExtC.cpp
 
 libTAO_CosNotification_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosNotification_la_LIBADD = \
   libTAO_CosEvent.la \
@@ -2015,7 +2015,7 @@
   ETCL/ETCL_y.cpp
 
 libTAO_ETCL_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_ETCL_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_AnyTypeCode.la \
@@ -2071,7 +2071,7 @@
   NotifyExtS.cpp
 
 libTAO_CosNotification_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosNotification_Skel_la_LIBADD = \
   libTAO_CosEvent_Skel.la \
@@ -2214,7 +2214,7 @@
   Notify/Worker_Task.cpp
 
 libTAO_CosNotification_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosNotification_Serv_la_LIBADD = \
   libTAO_ETCL.la \
@@ -2388,7 +2388,7 @@
   Notify/XML_Topology_Factory.cpp
 
 libTAO_CosNotification_Persist_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 if BUILD_ZLIB
 if BUILD_ZZIP
@@ -2481,7 +2481,7 @@
   TimeServiceS.cpp
 
 libTAO_CosTime_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosTime_la_LIBADD = \
   libTAO_Svc_Utils.la \
@@ -2620,7 +2620,7 @@
   CosTradingReposC.cpp
 
 libTAO_CosTrading_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosTrading_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_AnyTypeCode.la \
@@ -2673,7 +2673,7 @@
   CosTradingS.cpp
 
 libTAO_CosTrading_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosTrading_Skel_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PortableServer.la \
@@ -2740,7 +2740,7 @@
   Trader/Trading_Loader.cpp
 
 libTAO_CosTrading_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CosTrading_Serv_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_IORTable.la \
@@ -2868,7 +2868,7 @@
   DsLogNotificationC.cpp
 
 libTAO_DsLogAdmin_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsLogAdmin_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_AnyTypeCode.la \
@@ -2950,7 +2950,7 @@
   DsEventLogAdminC.cpp
 
 libTAO_DsEventLogAdmin_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsEventLogAdmin_la_LIBADD = \
   libTAO_CosEvent.la \
@@ -3000,7 +3000,7 @@
   DsLogNotificationS.cpp
 
 libTAO_DsLogAdmin_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsLogAdmin_Skel_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PortableServer.la \
@@ -3065,7 +3065,7 @@
   Log/Log_i.cpp
 
 libTAO_DsLogAdmin_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsLogAdmin_Serv_la_LIBADD = \
   libTAO_Svc_Utils.la \
@@ -3138,7 +3138,7 @@
   DsEventLogAdminS.cpp
 
 libTAO_DsEventLogAdmin_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsEventLogAdmin_Skel_la_LIBADD = \
   libTAO_CosEvent_Skel.la \
@@ -3195,7 +3195,7 @@
   Log/EventLog_i.cpp
 
 libTAO_DsEventLogAdmin_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsEventLogAdmin_Serv_la_LIBADD = \
   libTAO_CosEvent_Serv.la \
@@ -3297,7 +3297,7 @@
   DsNotifyLogAdminC.cpp
 
 libTAO_DsNotifyLogAdmin_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsNotifyLogAdmin_la_LIBADD = \
   libTAO_CosNotification.la \
@@ -3349,7 +3349,7 @@
   DsNotifyLogAdminS.cpp
 
 libTAO_DsNotifyLogAdmin_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsNotifyLogAdmin_Skel_la_LIBADD = \
   libTAO_CosNotification_Skel.la \
@@ -3409,7 +3409,7 @@
   Log/NotifyLog_i.cpp
 
 libTAO_DsNotifyLogAdmin_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DsNotifyLogAdmin_Serv_la_LIBADD = \
   libTAO_DsLogAdmin_Serv.la \
@@ -3502,7 +3502,7 @@
   FaultTolerance/FT_IOGR_Property.cpp
 
 libTAO_FTORB_Utils_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_FTORB_Utils_la_LIBADD = \
   libTAO_PortableGroup.la \
@@ -3566,7 +3566,7 @@
   FtRtEvent/ClientORB/FTRT_ClientORB_Loader.cpp
 
 libTAO_FTRT_ClientORB_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_FTRT_ClientORB_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_Messaging.la \
@@ -3685,7 +3685,7 @@
   RtecUDPAdminC.cpp
 
 libTAO_RTEvent_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTEvent_la_LIBADD = \
   libTAO_Svc_Utils.la \
@@ -3751,7 +3751,7 @@
   RtecUDPAdminS.cpp
 
 libTAO_RTEvent_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTEvent_Skel_la_LIBADD = \
   libTAO_RTEvent.la \
@@ -3884,7 +3884,7 @@
   Event/EC_UDP_Admin.cpp
 
 libTAO_RTEvent_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTEvent_Serv_la_LIBADD = \
   libTAO_RTEvent_Skel.la \
@@ -4155,7 +4155,7 @@
   FtRtecEventCommS.cpp
 
 libTAO_FtRtEvent_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_FtRtEvent_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PI_Server.la \
@@ -4264,7 +4264,7 @@
   FtRtEvent/EventChannel/replace_key.cpp
 
 libTAO_FTRT_EventChannel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_FTRT_EventChannel_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_Utils.la \
@@ -4364,7 +4364,7 @@
   FaultTolerance/FT_Service_Callbacks.cpp
 
 libTAO_FT_ClientORB_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_FT_ClientORB_la_LIBADD = \
   libTAO_FTORB_Utils.la \
@@ -4435,7 +4435,7 @@
   FaultTolerance/FT_ServerService_Activate.cpp
 
 libTAO_FT_ServerORB_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_FT_ServerORB_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PI_Server.la \
@@ -4625,7 +4625,7 @@
   FT_ReplicationManagerS.cpp
 
 libTAO_FaultTolerance_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_FaultTolerance_la_LIBADD = \
   libTAO_FT_ServerORB.la \
@@ -4762,7 +4762,7 @@
   HTIOPC.cpp
 
 libTAO_HTIOP_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_HTIOP_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_AnyTypeCode.la \
@@ -4870,7 +4870,7 @@
   IFRService/WstringDef_i.cpp
 
 libTAO_IFRService_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_IFRService_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_IORTable.la \
@@ -4991,7 +4991,7 @@
   Event/EC_RTCORBA_Factory.cpp
 
 libTAO_RTCORBAEvent_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTCORBAEvent_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_RTPortableServer.la \
@@ -5072,7 +5072,7 @@
   RTEventLogAdminC.cpp
 
 libTAO_RTEventLogAdmin_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTEventLogAdmin_la_LIBADD = \
   libTAO_RTEvent.la \
@@ -5161,7 +5161,7 @@
   Scheduler_Utilities.cpp
 
 libTAO_RTSched_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTSched_la_LIBADD = \
   libTAO_Svc_Utils.la \
@@ -5219,7 +5219,7 @@
   RTEventLogAdminS.cpp
 
 libTAO_RTEventLogAdmin_Skel_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTEventLogAdmin_Skel_la_LIBADD = \
   libTAO_RTEvent_Skel.la \
@@ -5282,7 +5282,7 @@
   Log/RTEventLog_i.cpp
 
 libTAO_RTEventLogAdmin_Serv_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTEventLogAdmin_Serv_la_LIBADD = \
   libTAO_RTSched.la \
@@ -5358,7 +5358,7 @@
   Event/EC_Kokyu_Scheduling.cpp
 
 libTAO_RTKokyuEvent_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTKokyuEvent_la_LIBADD = \
   $(ACE_BUILDDIR)/Kokyu/libKokyu.la \
@@ -5418,7 +5418,7 @@
   Event/EC_Sched_Filter_Builder.cpp
 
 libTAO_RTSchedEvent_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTSchedEvent_la_LIBADD = \
   libTAO_RTSched.la \
@@ -5481,7 +5481,7 @@
   Notify/Structured/RT_StructuredProxyPushSupplier.cpp
 
 libTAO_RT_Notification_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RT_Notification_la_LIBADD = \
   libTAO_CosNotification_Serv.la \
@@ -5707,7 +5707,7 @@
   SecurityS.cpp
 
 libTAO_Security_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Security_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_Valuetype.la \
@@ -5863,7 +5863,7 @@
   SSLIOPC.cpp
 
 libTAO_SSLIOP_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@ @ACE_TLS_LDFLAGS@
+ -release @TAO_VERSION_NAME@ @ACE_TLS_LDFLAGS@
 
 libTAO_SSLIOP_la_LIBADD = \
   $(TAO_BUILDDIR)/tao/libTAO_PI_Server.la \
diff -Nru ACE_wrappers.orig/TAO/tao/Makefile.am ACE_wrappers/TAO/tao/Makefile.am
--- ACE_wrappers.orig/TAO/tao/Makefile.am	2007-01-09 03:46:52.000000000 +0000
+++ ACE_wrappers/TAO/tao/Makefile.am	2007-02-07 12:46:13.000000000 +0000
@@ -954,7 +954,7 @@
   target_specification.cpp
 
 libTAO_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_la_LIBADD = \
   $(ACE_BUILDDIR)/ace/libACE.la
@@ -1535,7 +1535,7 @@
   Codeset/UTF8_Latin1_Translator.cpp
 
 libTAO_Codeset_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Codeset_la_LIBADD = \
   libTAO.la \
@@ -1760,7 +1760,7 @@
   AnyTypeCode/skip.cpp
 
 libTAO_AnyTypeCode_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_AnyTypeCode_la_LIBADD = \
   libTAO.la \
@@ -2004,7 +2004,7 @@
   CodecFactory/IOP_Codec_includeC.cpp
 
 libTAO_CodecFactory_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CodecFactory_la_LIBADD = \
   libTAO_AnyTypeCode.la \
@@ -2304,7 +2304,7 @@
   PI/RequestInfo_Util.cpp
 
 libTAO_PI_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_PI_la_LIBADD = \
   libTAO_CodecFactory.la \
@@ -2446,7 +2446,7 @@
   BiDir_GIOP/BiDir_Policy_i.cpp
 
 libTAO_BiDirGIOP_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_BiDirGIOP_la_LIBADD = \
   libTAO_PI.la \
@@ -2979,7 +2979,7 @@
   PortableServer/Upcall_Wrapper.cpp
 
 libTAO_PortableServer_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_PortableServer_la_LIBADD = \
   libTAO_AnyTypeCode.la \
@@ -3302,7 +3302,7 @@
   CSD_Framework/CSD_Strategy_Repository.cpp
 
 libTAO_CSD_Framework_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CSD_Framework_la_LIBADD = \
   libTAO_PI.la \
@@ -3388,7 +3388,7 @@
   CSD_ThreadPool/CSD_ThreadPool.cpp
 
 libTAO_CSD_ThreadPool_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_CSD_ThreadPool_la_LIBADD = \
   libTAO_CSD_Framework.la \
@@ -3496,7 +3496,7 @@
   Compression/Compressor_Factory.cpp
 
 libTAO_Compression_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Compression_la_LIBADD = \
   libTAO.la \
@@ -3634,7 +3634,7 @@
   IFR_Client/IFR_ExtendedC.cpp
 
 libTAO_IFR_Client_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_IFR_Client_la_LIBADD = \
   libTAO_AnyTypeCode.la \
@@ -3702,7 +3702,7 @@
   Domain/DomainS.cpp
 
 libTAO_Domain_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Domain_la_LIBADD = \
   libTAO_PortableServer.la \
@@ -3778,7 +3778,7 @@
   Valuetype/Valuetype_Adapter_Impl.cpp
 
 libTAO_Valuetype_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Valuetype_la_LIBADD = \
   libTAO_AnyTypeCode.la \
@@ -3876,7 +3876,7 @@
   DynamicAny/DynamicAnyC.cpp
 
 libTAO_DynamicAny_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DynamicAny_la_LIBADD = \
   libTAO_Valuetype.la \
@@ -4133,7 +4133,7 @@
   Messaging/TAO_ExtC.cpp
 
 libTAO_Messaging_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Messaging_la_LIBADD = \
   libTAO_Valuetype.la \
@@ -4239,7 +4239,7 @@
   DynamicInterface/Unknown_User_Exception.cpp
 
 libTAO_DynamicInterface_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_DynamicInterface_la_LIBADD = \
   libTAO_Messaging.la \
@@ -4382,7 +4382,7 @@
   EndpointPolicy/IIOPEndpointValue_i.cpp
 
 libTAO_EndpointPolicy_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_EndpointPolicy_la_LIBADD = \
   libTAO_PI.la \
@@ -4516,7 +4516,7 @@
   ObjRefTemplate/ObjectReferenceTemplate_includeC.cpp
 
 libTAO_ObjRefTemplate_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_ObjRefTemplate_la_LIBADD = \
   libTAO_PortableServer.la \
@@ -4621,7 +4621,7 @@
   IORInterceptor/IORInterceptor_Details.cpp
 
 libTAO_IORInterceptor_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_IORInterceptor_la_LIBADD = \
   libTAO_PI.la \
@@ -4704,7 +4704,7 @@
   IORManipulation/IORManipulation.cpp
 
 libTAO_IORManip_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_IORManip_la_LIBADD = \
   libTAO_AnyTypeCode.la \
@@ -4774,7 +4774,7 @@
   IORTable/Table_Adapter.cpp
 
 libTAO_IORTable_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_IORTable_la_LIBADD = \
   libTAO.la \
@@ -4877,7 +4877,7 @@
   ImR_Client/ServerObject_i.cpp
 
 libTAO_ImR_Client_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_ImR_Client_la_LIBADD = \
   libTAO_PortableServer.la \
@@ -4988,7 +4988,7 @@
   PI_Server/ServerRequestInterceptor_Factory_Impl.cpp
 
 libTAO_PI_Server_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_PI_Server_la_LIBADD = \
   libTAO_PortableServer.la \
@@ -5108,7 +5108,7 @@
   RTCORBA/Thread_Pool.cpp
 
 libTAO_RTCORBA_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTCORBA_la_LIBADD = \
   libTAO_PI.la \
@@ -5243,7 +5243,7 @@
   RTPortableServer/RT_Servant_Dispatcher.cpp
 
 libTAO_RTPortableServer_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTPortableServer_la_LIBADD = \
   libTAO_PortableServer.la \
@@ -5357,7 +5357,7 @@
   RTScheduling/Request_Interceptor.cpp
 
 libTAO_RTScheduler_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_RTScheduler_la_LIBADD = \
   libTAO_PI_Server.la \
@@ -5426,7 +5426,7 @@
   SmartProxies/Smart_Proxies.cpp
 
 libTAO_SmartProxies_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_SmartProxies_la_LIBADD = \
   libTAO.la \
@@ -5568,7 +5568,7 @@
   Strategies/uiop_endpointsC.cpp
 
 libTAO_Strategies_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Strategies_la_LIBADD = \
   libTAO_AnyTypeCode.la \
@@ -5679,7 +5679,7 @@
   FlResource/FlResource_Loader.cpp
 
 libTAO_FlResource_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@ $(ACE_FLTK_LDFLAGS)
+ -release @TAO_VERSION_NAME@ $(ACE_FLTK_LDFLAGS)
 
 libTAO_FlResource_la_LIBADD = \
   $(ACE_BUILDDIR)/ace/libACE_FlReactor.la \
@@ -5719,7 +5719,7 @@
   QtResource/QtResource_Loader.cpp
 
 libTAO_QtResource_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@ $(ACE_QT_LDFLAGS)
+ -release @TAO_VERSION_NAME@ $(ACE_QT_LDFLAGS)
 
 libTAO_QtResource_la_LIBADD = \
   $(ACE_BUILDDIR)/ace/libACE_QtReactor.la \
@@ -5758,7 +5758,7 @@
   TkResource/TkResource_Loader.cpp
 
 libTAO_TkResource_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@ $(ACE_TK_LDFLAGS) $(ACE_TCL_LDFLAGS)
+ -release @TAO_VERSION_NAME@ $(ACE_TK_LDFLAGS) $(ACE_TCL_LDFLAGS)
 
 libTAO_TkResource_la_LIBADD = \
   $(ACE_BUILDDIR)/ace/libACE_TkReactor.la \
@@ -5799,7 +5799,7 @@
   XtResource/XtResource_Loader.cpp
 
 libTAO_XtResource_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@ $(ACE_X11_LDFLAGS) $(ACE_XT_LDFLAGS)
+ -release @TAO_VERSION_NAME@ $(ACE_X11_LDFLAGS) $(ACE_XT_LDFLAGS)
 
 libTAO_XtResource_la_LIBADD = \
   $(ACE_BUILDDIR)/ace/libACE_XtReactor.la \
@@ -5863,7 +5863,7 @@
   TransportCurrent/TCS.cpp
 
 libTAO_TC_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_TC_la_LIBADD = \
   libTAO_PI.la \
@@ -5932,7 +5932,7 @@
   TransportCurrent/TC_IIOPS.cpp
 
 libTAO_TC_IIOP_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_TC_IIOP_la_LIBADD = \
   libTAO_TC.la \
@@ -5993,7 +5993,7 @@
   TypeCodeFactory/TypeCodeFactory_i.cpp
 
 libTAO_TypeCodeFactory_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_TypeCodeFactory_la_LIBADD = \
   libTAO_IFR_Client.la \
@@ -6053,7 +6053,7 @@
   Utils/Synch_Refcountable.cpp
 
 libTAO_Utils_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@
+ -release @TAO_VERSION_NAME@
 
 libTAO_Utils_la_LIBADD = \
   libTAO_PI.la \
@@ -6114,7 +6114,7 @@
   Compression/zlib/ZlibCompressor_Factory.cpp
 
 libTAO_ZlibCompressor_la_LDFLAGS = \
- -version-number @TAO_MAJOR@:@TAO_MINOR@:@TAO_BETA@ $(ACE_ZLIB_LDFLAGS)
+ -release @TAO_VERSION_NAME@ $(ACE_ZLIB_LDFLAGS)
 
 libTAO_ZlibCompressor_la_LIBADD = \
   libTAO_Compression.la \
