--- pkg-ace.orig/etc/ace.doxygen
+++ pkg-ace/etc/ace.doxygen
@@ -89,7 +89,7 @@
 HTML_FOOTER           =
 HTML_STYLESHEET       =
 HTML_ALIGN_MEMBERS    = YES
-GENERATE_HTMLHELP     = YES
+GENERATE_HTMLHELP     = NO
 
 #---------------------------------------------------------------------------
 # configuration options related to the LaTeX output
@@ -174,11 +174,13 @@
 
 HAVE_DOT              = YES
 CLASS_GRAPH           = YES
-COLLABORATION_GRAPH   = YES
+COLLABORATION_GRAPH   = NO
 INCLUDE_GRAPH         = NO
 INCLUDED_BY_GRAPH     = NO
 GRAPHICAL_HIERARCHY   = YES
-DOT_MULTI_TARGETS     = YES
+DOT_MULTI_TARGETS     = YES 
+DOT_GRAPH_MAX_NODES   = 15 
+DOT_IMAGE_FORMAT      = png
 MAX_DOT_GRAPH_DEPTH   = 3
 
 # The tag DOT_PATH can be used to specify the path where the dot tool can be
@@ -189,4 +191,4 @@
 # Configuration::additions related to the search engine
 #---------------------------------------------------------------------------
 
-SEARCHENGINE          = YES
+SEARCHENGINE          = NO
--- pkg-ace.orig/etc/ace_inet.doxygen
+++ pkg-ace/etc/ace_inet.doxygen
@@ -79,7 +79,7 @@
 HTML_FOOTER           =
 HTML_STYLESHEET       =
 HTML_ALIGN_MEMBERS    = YES
-GENERATE_HTMLHELP     = YES
+GENERATE_HTMLHELP     = NO
 
 #---------------------------------------------------------------------------
 # configuration options related to the LaTeX output
@@ -147,11 +147,13 @@
 
 HAVE_DOT              = YES
 CLASS_GRAPH           = YES
-COLLABORATION_GRAPH   = YES
+COLLABORATION_GRAPH   = NO
 INCLUDE_GRAPH         = NO
 INCLUDED_BY_GRAPH     = NO
 GRAPHICAL_HIERARCHY   = YES
-DOT_MULTI_TARGETS     = YES
+DOT_MULTI_TARGETS     = YES 
+DOT_GRAPH_MAX_NODES   = 15 
+DOT_IMAGE_FORMAT      = png
 MAX_DOT_GRAPH_DEPTH   = 3
 
 # The tag DOT_PATH can be used to specify the path where the dot tool can be
@@ -162,5 +164,5 @@
 # Configuration::additions related to the search engine
 #---------------------------------------------------------------------------
 
-SEARCHENGINE          = YES
+SEARCHENGINE          = NO
 
--- pkg-ace.orig/etc/ace_man.doxygen
+++ pkg-ace/etc/ace_man.doxygen
@@ -79,7 +79,7 @@
 HTML_FOOTER           =
 HTML_STYLESHEET       =
 HTML_ALIGN_MEMBERS    = YES
-GENERATE_HTMLHELP     = YES
+GENERATE_HTMLHELP     = NO
 
 #---------------------------------------------------------------------------
 # configuration options related to the LaTeX output
@@ -151,7 +151,9 @@
 INCLUDE_GRAPH         = NO
 INCLUDED_BY_GRAPH     = NO
 GRAPHICAL_HIERARCHY   = NO
-DOT_MULTI_TARGETS     = YES
+DOT_MULTI_TARGETS     = YES 
+DOT_GRAPH_MAX_NODES   = 15 
+DOT_IMAGE_FORMAT      = png
 MAX_DOT_GRAPH_DEPTH   = 3
 
 # The tag DOT_PATH can be used to specify the path where the dot tool can be
--- pkg-ace.orig/etc/ace_qos.doxygen
+++ pkg-ace/etc/ace_qos.doxygen
@@ -79,7 +79,7 @@
 HTML_FOOTER           =
 HTML_STYLESHEET       =
 HTML_ALIGN_MEMBERS    = YES
-GENERATE_HTMLHELP     = YES
+GENERATE_HTMLHELP     = NO
 
 #---------------------------------------------------------------------------
 # configuration options related to the LaTeX output
@@ -147,11 +147,13 @@
 
 HAVE_DOT              = YES
 CLASS_GRAPH           = YES
-COLLABORATION_GRAPH   = YES
+COLLABORATION_GRAPH   = NO
 INCLUDE_GRAPH         = NO
 INCLUDED_BY_GRAPH     = NO
 GRAPHICAL_HIERARCHY   = YES
-DOT_MULTI_TARGETS     = YES
+DOT_MULTI_TARGETS     = YES 
+DOT_GRAPH_MAX_NODES   = 15 
+DOT_IMAGE_FORMAT      = png
 MAX_DOT_GRAPH_DEPTH   = 3
 
 # The tag DOT_PATH can be used to specify the path where the dot tool can be
@@ -162,5 +164,5 @@
 # Configuration::additions related to the search engine
 #---------------------------------------------------------------------------
 
-SEARCHENGINE          = YES
+SEARCHENGINE          = NO
 
--- pkg-ace.orig/etc/ace_rmcast.doxygen
+++ pkg-ace/etc/ace_rmcast.doxygen
@@ -79,7 +79,7 @@
 HTML_FOOTER           =
 HTML_STYLESHEET       =
 HTML_ALIGN_MEMBERS    = YES
-GENERATE_HTMLHELP     = YES
+GENERATE_HTMLHELP     = NO
 
 #---------------------------------------------------------------------------
 # configuration options related to the LaTeX output
@@ -147,11 +147,13 @@
 
 HAVE_DOT              = YES
 CLASS_GRAPH           = YES
-COLLABORATION_GRAPH   = YES
+COLLABORATION_GRAPH   = NO
 INCLUDE_GRAPH         = NO
 INCLUDED_BY_GRAPH     = NO
 GRAPHICAL_HIERARCHY   = YES
-DOT_MULTI_TARGETS     = YES
+DOT_MULTI_TARGETS     = YES 
+DOT_GRAPH_MAX_NODES   = 15 
+DOT_IMAGE_FORMAT      = png
 MAX_DOT_GRAPH_DEPTH   = 3
 
 # The tag DOT_PATH can be used to specify the path where the dot tool can be
@@ -162,5 +164,5 @@
 # Configuration::additions related to the search engine
 #---------------------------------------------------------------------------
 
-SEARCHENGINE          = YES
+SEARCHENGINE          = NO
 
--- pkg-ace.orig/etc/ace_ssl.doxygen
+++ pkg-ace/etc/ace_ssl.doxygen
@@ -79,7 +79,7 @@
 HTML_FOOTER           =
 HTML_STYLESHEET       =
 HTML_ALIGN_MEMBERS    = YES
-GENERATE_HTMLHELP     = YES
+GENERATE_HTMLHELP     = NO
 
 #---------------------------------------------------------------------------
 # configuration options related to the LaTeX output
@@ -147,11 +147,13 @@
 
 HAVE_DOT              = YES
 CLASS_GRAPH           = YES
-COLLABORATION_GRAPH   = YES
+COLLABORATION_GRAPH   = NO
 INCLUDE_GRAPH         = NO
 INCLUDED_BY_GRAPH     = NO
 GRAPHICAL_HIERARCHY   = YES
-DOT_MULTI_TARGETS     = YES
+DOT_MULTI_TARGETS     = YES 
+DOT_GRAPH_MAX_NODES   = 15 
+DOT_IMAGE_FORMAT      = png
 MAX_DOT_GRAPH_DEPTH   = 3
 
 # The tag DOT_PATH can be used to specify the path where the dot tool can be
@@ -162,5 +164,5 @@
 # Configuration::additions related to the search engine
 #---------------------------------------------------------------------------
 
-SEARCHENGINE          = YES
+SEARCHENGINE          = NO
 
--- pkg-ace.orig/etc/acexml.doxygen
+++ pkg-ace/etc/acexml.doxygen
@@ -79,7 +79,7 @@
 HTML_FOOTER           =
 HTML_STYLESHEET       =
 HTML_ALIGN_MEMBERS    = YES
-GENERATE_HTMLHELP     = YES
+GENERATE_HTMLHELP     = NO
 
 #---------------------------------------------------------------------------
 # configuration options related to the LaTeX output
@@ -147,11 +147,13 @@
 
 HAVE_DOT              = YES
 CLASS_GRAPH           = YES
-COLLABORATION_GRAPH   = YES
+COLLABORATION_GRAPH   = NO
 INCLUDE_GRAPH         = NO
 INCLUDED_BY_GRAPH     = NO
 GRAPHICAL_HIERARCHY   = YES
-DOT_MULTI_TARGETS     = YES
+DOT_MULTI_TARGETS     = YES 
+DOT_GRAPH_MAX_NODES   = 15 
+DOT_IMAGE_FORMAT      = png
 MAX_DOT_GRAPH_DEPTH   = 3
 
 # The tag DOT_PATH can be used to specify the path where the dot tool can be
@@ -162,5 +164,5 @@
 # Configuration::additions related to the search engine
 #---------------------------------------------------------------------------
 
-SEARCHENGINE          = YES
+SEARCHENGINE          = NO
 
