--- ace-5.4.orig/ACEXML/Makefile
+++ ace-5.4/ACEXML/Makefile
@@ -8,12 +8,12 @@
 MFILES = \
          common/Makefile.ACEXML \
          parser/parser/Makefile.ACEXML_Parser \
-         apps/svcconf/Makefile.ACEXML_XML_Svc_Conf_Parser \
-         examples/SAXPrint/Makefile.SAXPrint \
-         tests/Makefile.Transcoder_Test \
-         tests/Makefile.NamespaceSupport_Test \
-         tests/Makefile.HttpCharStream_Test \
-         tests/Makefile.ContentHandler_Test
+         apps/svcconf/Makefile.ACEXML_XML_Svc_Conf_Parser
+#         examples/SAXPrint/Makefile.SAXPrint \
+#         tests/Makefile.Transcoder_Test \
+#         tests/Makefile.NamespaceSupport_Test \
+#         tests/Makefile.HttpCharStream_Test \
+#         tests/Makefile.ContentHandler_Test
 
 $(TARGETS_NESTED):
 	@for file in $(MFILES); do \
--- ace-5.4.orig/TAO/CIAO/Makefile
+++ ace-5.4/TAO/CIAO/Makefile
@@ -18,9 +18,9 @@
 
 INFO    = VERSION
 
-ALT_DIRS = CIDLC \
-           examples \
-           tests
+ALT_DIRS = CIDLC 
+#           examples \
+#           tests
 
 CLONE   = Makefile \
 	  CIDLC \
--- ace-5.4.orig/TAO/CIAO/examples/Hello/descriptors/run_test_default.pl
+++ ace-5.4/TAO/CIAO/examples/Hello/descriptors/run_test_default.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/CIAO/examples/Hello/descriptors/run_test_remote.pl
+++ ace-5.4/TAO/CIAO/examples/Hello/descriptors/run_test_remote.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 #run_test_remote.pl,v 1.1 2003/12/03 00:46:12 lut Exp
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
--- ace-5.4.orig/TAO/Makefile
+++ ace-5.4/TAO/Makefile
@@ -21,23 +21,19 @@
 
 DIRS    = tao \
           TAO_IDL \
-          tests \
           orbsvcs \
-          examples \
-          performance-tests \
           utils \
-	  docs/tutorials/Quoter \
           CIAO
+#          tests \
+#          examples \
+#          performance-tests \
+#	  docs/tutorials/Quoter \
 
 CLONE   = Makefile \
           tao \
           TAO_IDL \
-          tests \
           orbsvcs \
-          examples \
-          performance-tests \
-          utils \
-	  docs/tutorials/Quoter
+          utils 
 
 #----------------------------------------------------------------------------
 #       Include macros and targets
--- ace-5.4.orig/TAO/TAO_IDL/util/utl_global.cpp
+++ ace-5.4/TAO/TAO_IDL/util/utl_global.cpp
@@ -198,6 +198,7 @@
       const char ace_gperf[] = ACE_GPERF;
       ACE_NEW (this->gperf_path_,
                char [ACE_OS::strlen (ace_root)
+                     + ACE_OS::strlen ("/bin/")
                      + ACE_OS::strlen (ace_gperf)
                      + 1]);
 #  if defined (ACE_WIN32)
--- ace-5.4.orig/TAO/examples/AMH/Sink_Server/run_test.pl
+++ ace-5.4/TAO/examples/AMH/Sink_Server/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Advanced/ch_12/run_test.pl
+++ ace-5.4/TAO/examples/Advanced/ch_12/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Advanced/ch_18/run_test.pl
+++ ace-5.4/TAO/examples/Advanced/ch_18/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Advanced/ch_21/run_test.pl
+++ ace-5.4/TAO/examples/Advanced/ch_21/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Advanced/ch_3/run_test.pl
+++ ace-5.4/TAO/examples/Advanced/ch_3/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Advanced/ch_8_and_10/run_test.pl
+++ ace-5.4/TAO/examples/Advanced/ch_8_and_10/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Advanced/run_test.pl
+++ ace-5.4/TAO/examples/Advanced/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Buffered_AMI/run_test.pl
+++ ace-5.4/TAO/examples/Buffered_AMI/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Buffered_Oneways/run_test.pl
+++ ace-5.4/TAO/examples/Buffered_Oneways/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Callback_Quoter/run_test.pl
+++ ace-5.4/TAO/examples/Callback_Quoter/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Load_Balancing/run_test.pl
+++ ace-5.4/TAO/examples/Load_Balancing/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Logging/run_test.pl
+++ ace-5.4/TAO/examples/Logging/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/OBV/Typed_Events/run_test.pl
+++ ace-5.4/TAO/examples/OBV/Typed_Events/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/Adapter_Activator/run_test.pl
+++ ace-5.4/TAO/examples/POA/Adapter_Activator/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/DSI/run_test.pl
+++ ace-5.4/TAO/examples/POA/DSI/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/Default_Servant/run_test.pl
+++ ace-5.4/TAO/examples/POA/Default_Servant/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/Explicit_Activation/run_test.pl
+++ ace-5.4/TAO/examples/POA/Explicit_Activation/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/FindPOA/run_test.pl
+++ ace-5.4/TAO/examples/POA/FindPOA/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/Forwarding/run_test.pl
+++ ace-5.4/TAO/examples/POA/Forwarding/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/Loader/run_test.pl
+++ ace-5.4/TAO/examples/POA/Loader/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/NewPOA/run_test.pl
+++ ace-5.4/TAO/examples/POA/NewPOA/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/On_Demand_Activation/run_test.pl
+++ ace-5.4/TAO/examples/POA/On_Demand_Activation/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/On_Demand_Loading/run_test.pl
+++ ace-5.4/TAO/examples/POA/On_Demand_Loading/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/POA_BiDir/run_test.pl
+++ ace-5.4/TAO/examples/POA/POA_BiDir/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/Reference_Counted_Servant/run_test.pl
+++ ace-5.4/TAO/examples/POA/Reference_Counted_Servant/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/RootPOA/run_test.pl
+++ ace-5.4/TAO/examples/POA/RootPOA/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/POA/TIE/run_test.pl
+++ ace-5.4/TAO/examples/POA/TIE/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Persistent_Grid/run_test.pl
+++ ace-5.4/TAO/examples/Persistent_Grid/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/PluggableUDP/tests/Basic/run_test.pl
+++ ace-5.4/TAO/examples/PluggableUDP/tests/Basic/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/PluggableUDP/tests/Performance/run_test.pl
+++ ace-5.4/TAO/examples/PluggableUDP/tests/Performance/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/PluggableUDP/tests/SimplePerformance/run_test.pl
+++ ace-5.4/TAO/examples/PluggableUDP/tests/SimplePerformance/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Quoter/run_test.pl
+++ ace-5.4/TAO/examples/Quoter/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/RTCORBA/Activity/run_test.pl
+++ ace-5.4/TAO/examples/RTCORBA/Activity/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/RTScheduling/Fixed_Priority_Scheduler/run_test.pl
+++ ace-5.4/TAO/examples/RTScheduling/Fixed_Priority_Scheduler/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/RTScheduling/MIF_Scheduler/run_test.pl
+++ ace-5.4/TAO/examples/RTScheduling/MIF_Scheduler/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Simple/bank/run_test.pl
+++ ace-5.4/TAO/examples/Simple/bank/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Simple/echo/run_test.pl
+++ ace-5.4/TAO/examples/Simple/echo/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Simple/grid/run_test.pl
+++ ace-5.4/TAO/examples/Simple/grid/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Simple/time/run_test.pl
+++ ace-5.4/TAO/examples/Simple/time/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Simple/time-date/run_test.pl
+++ ace-5.4/TAO/examples/Simple/time-date/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/Simulator/DOVEBrowser/ecdata_patch.pl
+++ ace-5.4/TAO/examples/Simulator/DOVEBrowser/ecdata_patch.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -pi -S $0 ${1+"$@"}'
     & eval 'exec perl -pi -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/TypeCode_Creation/run_test.pl
+++ ace-5.4/TAO/examples/TypeCode_Creation/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/examples/ior_corbaloc/run_test.pl
+++ ace-5.4/TAO/examples/ior_corbaloc/run_test.pl
@@ -1,4 +1,6 @@
-     eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
+#!/usr/bin/perl
+
+eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
   &  eval 'exec perl -S $0 $argv:q'
   if 0;
 
--- ace-5.4.orig/TAO/orbsvcs/Makefile
+++ ace-5.4/TAO/orbsvcs/Makefile
@@ -43,10 +43,10 @@
 	Trading_Service \
 	LoadBalancer \
 	TAO_Service \
-	PSS \
-        tests \
-	performance-tests \
-        examples
+	PSS
+#        tests \
+#	performance-tests \
+#        examples
 
 ifeq ($(minimum_corba),0)
     ifneq (Compilers 4.2,$(findstring Compilers 4.2,$(CC_VERSION)))
--- ace-5.4.orig/TAO/orbsvcs/Scheduling_Service/Makefile
+++ ace-5.4/TAO/orbsvcs/Scheduling_Service/Makefile
@@ -23,7 +23,7 @@
 #### If the TAO orbsvcs library wasn't built with sufficient components,
 #### don't try to build here.
 TAO_ORBSVCS := $(shell sh $(ACE_ROOT)/bin/ace_components --orbsvcs)
-ifeq (RtSched,$(findstring RTSched,$(TAO_ORBSVCS)))
+ifeq (RTSched,$(findstring RTSched,$(TAO_ORBSVCS)))
   BIN = $(BIN2)
 endif # Sched
 include $(ACE_ROOT)/include/makeinclude/rules.common.GNU
--- ace-5.4.orig/TAO/orbsvcs/examples/CosEC/RtEC_Based/tests/Multiple/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/CosEC/RtEC_Based/tests/Multiple/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/CosEC/RtEC_Based/tests/Multiple/run_test2.pl
+++ ace-5.4/TAO/orbsvcs/examples/CosEC/RtEC_Based/tests/Multiple/run_test2.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/CosEC/Simple/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/CosEC/Simple/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/LoadBalancing/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/LoadBalancing/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/Notify/Filter/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/Notify/Filter/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/Notify/Lanes/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/Notify/Lanes/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/Notify/Subscribe/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/Notify/Subscribe/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/Notify/ThreadPool/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/Notify/ThreadPool/run_test.pl
@@ -1,3 +1,4 @@
+#!/usr/bin/perl
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/RtEC/IIOPGateway/consumerec_crash.pl
+++ ace-5.4/TAO/orbsvcs/examples/RtEC/IIOPGateway/consumerec_crash.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/RtEC/IIOPGateway/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/RtEC/IIOPGateway/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/RtEC/Simple/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/RtEC/Simple/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/examples/Security/Send_File/run_test.pl
+++ ace-5.4/TAO/orbsvcs/examples/Security/Send_File/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_ClientScheduler_i.cpp
+++ ace-5.4/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_ClientScheduler_i.cpp
@@ -4,7 +4,7 @@
 /**
  *  @file    RTCosScheduling_ClientScheduler_i.cpp
  *
- *  RTCosScheduling_ClientScheduler_i.cpp,v 1.8 2003/12/07 18:28:11 bala Exp
+ *  $Id: RTCosScheduling_ClientScheduler_i.cpp,v 1.9 2004/02/13 21:18:09 bryank Exp $
  *
  *  @author Matt Murphy <murphym@cs.uri.edu>
  */
@@ -213,7 +213,7 @@
     }
 
 
-  char line[BUF_MAX], key[16];
+  char line[BUF_MAX], key[64];
   ACE_OS::strsncpy(key, "Node ", sizeof(key));
   ACE_OS::strcat(key, node_name);
 
@@ -222,9 +222,6 @@
   do
     {
       ACE_OS::fgets(line, BUF_MAX, fp);
-    }
-  while (ACE_OS::strncmp(line, key, ACE_OS::strlen(key)) != 0);
-
   /// Make sure we did not hit the end of file
   if (ACE_OS::last_error() == EOF)
     {
@@ -232,28 +229,31 @@
                         "Node %s not found in config file\n",
                         node_name),
                         0);
+          break;
+        }
     }
+  while (ACE_OS::strncmp(line, key, ACE_OS::strlen(key)) != 0);
 
   /// Skip to the appropriate task section of the node
   ACE_OS::clearerr(fp);
   do
     {
       ACE_OS::fgets(line, BUF_MAX, fp);
-    }
-  while (ACE_OS::strncmp(line, "Tasks:", ACE_OS::strlen("Tasks:")) != 0);
-
   /// Make sure we did not hit the end of file
-  if (ACE_OS::last_error() == EOF)
-    {
-      ACE_ERROR_RETURN((LM_ERROR,
-                        "Task list not found for node %s\n",
-                        node_name),
-                        0);
+      if (ACE_OS::last_error() == EOF)
+        {
+          ACE_ERROR_RETURN((LM_ERROR,
+                            "Task list not found for node %s\n",
+                            node_name),
+                            0);
+              break;
+        }
     }
+  while (ACE_OS::strncmp(line, "Tasks:", ACE_OS::strlen("Tasks:")) != 0);
 
   CORBA::Short done = 0;
   COS_SCHEDULER_ACTIVITY_KEY name;
-  COS_SCHEDULER_ACTIVITY_VALUE priority;
+  COS_SCHEDULER_ACTIVITY_VALUE priority = 0;
   u_int delimiter;
 
   /// read each activity/priority pair from the config file
@@ -262,12 +262,26 @@
       /// get the activity name
       ACE_OS::fgets(line, BUF_MAX, fp);
 
+      /// Make sure we did not hit the end of file
+      if (ACE_OS::last_error() == EOF)
+        {
+          ACE_ERROR_RETURN((LM_ERROR,
+                            "Task list not found for node %s\n",
+                            node_name),
+                            0);
+          break;
+        }
+
       /// check to make sure we have not reached the end of the list.
       if (ACE_OS::strncmp(line, "END", ACE_OS::strlen("END")) != 0)
         {
           name      = ACE_CString(line);
           delimiter = name.find('\t');
-          priority  = ACE_OS::atoi(ACE_OS::strchr(line, '\t'));
+          char *p = ACE_OS::strchr(line, '\t');
+          if (p)
+            priority  = ACE_OS::atoi(p);
+          if (priority == 0)
+              priority = RTCORBA::minPriority;
           if (delimiter < name.length() && delimiter > 0)
             {
               activity_map->bind(name.substr(0, delimiter), priority);
--- ace-5.4.orig/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_ClientScheduler_i.h
+++ ace-5.4/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_ClientScheduler_i.h
@@ -4,7 +4,7 @@
 /**
  *  @file    RTCosScheduling_ClientScheduler_i.h
  *
- *  RTCosScheduling_ClientScheduler_i.h,v 1.3 2003/10/28 18:34:24 bala Exp
+ *  $Id: RTCosScheduling_ClientScheduler_i.h,v 1.4 2004/02/04 20:20:23 parsons Exp $
  *
  *  @author Matt Murphy <murphym@cs.uri.edu>
  */
@@ -40,8 +40,8 @@
 #endif /* _MSC_VER */
 
 #include "ace/SString.h"
-#include <orbsvcs/RTCosSchedulingC.h>
-#include <orbsvcs/RTCosScheduling/RTCosScheduling_export.h>
+#include <orbsvcs/orbsvcs/RTCosSchedulingC.h>
+#include <orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_export.h>
 #include "tao/RTCORBA/Priority_Mapping_Manager.h"
 #include "tao/PortableInterceptorC.h"
 #include "tao/RTCORBA/Linear_Priority_Mapping.h"
--- ace-5.4.orig/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_PCP_Manager.cpp
+++ ace-5.4/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_PCP_Manager.cpp
@@ -4,7 +4,7 @@
 /**
  *  @file    RTCosScheduling_PCP_Manager.cpp
  *
- *  RTCosScheduling_PCP_Manager.cpp,v 1.6 2003/12/30 05:03:10 dhinton Exp
+ *  $Id: RTCosScheduling_PCP_Manager.cpp,v 1.7 2004/02/13 21:18:09 bryank Exp $
  *
  *  @author Matt Murphy <murphym@cs.uri.edu>
  */
@@ -23,50 +23,8 @@
 
 namespace TAO {
 
-struct
-CosSchedulingLockNode *CosSchedulingLockNode::next()
-{
-  /// INT_MAX is a special value indicating the end of a list
-  if (this->next_offset_ == INT_MAX)
-    {
-      return 0;
-    }
-  else
-    {
-      return ACE_reinterpret_cast(CosSchedulingLockNode *,
-               (ACE_reinterpret_cast(int, this) + this->next_offset_)
-             );
-    }
-}
-
-void
-CosSchedulingLockNode::next(struct CosSchedulingLockNode *next_lock)
-{
-  /// INT_MAX is a special value indicating the end of a list
-  if (next_lock == 0)
-    {
-      this->next_offset_ = INT_MAX;
-    }
-  else
-    {
-      this->next_offset_ =
-        (ACE_reinterpret_cast(int, next_lock) -
-         ACE_reinterpret_cast(int, this));
-    }
-}
-
-const CosSchedulingLockNode& CosSchedulingLockNode::operator=(const CosSchedulingLockNode& L)
-{
-  this->threadID_           = L.threadID_;
-  this->priority_ceiling_   = L.priority_ceiling_;
-  this->priority_           = L.priority_;
-  this->elevated_priority_  = L.elevated_priority_;
-
-  return *this;
-}
-
 CosSchedulingLockList::CosSchedulingLockList(CosSchedulingLockNode *lock_array,
-  int size,
+  const int size,
   ACE_SYNCH_MUTEX *mutex)
 {
   ACE_TRY_NEW_ENV
@@ -107,14 +65,18 @@
     }
   ACE_CATCHANY
     {
+      ACE_DEBUG((LM_ERROR,
+                 "Error in %s: Line %d - Could not generate a Locklist in shared memory\n",
+                 __FILE__,
+                 __LINE__));
       ACE_PRINT_EXCEPTION(ACE_ANY_EXCEPTION,
-                         "Error in generating Locklist on ServerScheduler\n");
+                          "Exception: CosSchedulingLockList()");
     }
   ACE_ENDTRY;
 }
 
 void
-CosSchedulingLockList::destroy(int size)
+CosSchedulingLockList::destroy(const int size)
 {
   for (int i = 3; i < size; ++i)
     {
@@ -270,17 +232,17 @@
   /// take pending lock off the head of the list
   /// (highest priority request) and add to the free list
   L = *(this->pending_->next());
-  L.next(this->pending_->next()->next());
-  this->pending_->next()->next(free_->next());
-  this->free_->next(this->pending_->next());
-  this->pending_->next(L.next());
-  L.next(0);
+  CosSchedulingLockNode * fn = this->pending_->next();
+  this->pending_->next(this->pending_->next()->next());
+  fn->next(this->free_->next());
+  this->free_->next(fn);
+
   return 1;
 }
 
 PCP_Manager::PCP_Manager(CosSchedulingLockList *locks,
   ACE_SYNCH_MUTEX *mutex,
-  RTCORBA::Current_var current)
+  const RTCORBA::Current_var current)
 : locks_(locks),
   mutex_(mutex),
   current_(current)
@@ -290,7 +252,7 @@
 }
 
 void
-PCP_Manager::lock(int priority_ceiling, int priority)
+PCP_Manager::lock(const int priority_ceiling, const int priority)
 {
   ACE_TRY_NEW_ENV
     {
@@ -375,8 +337,12 @@
     }
   ACE_CATCHANY
     {
+      ACE_DEBUG((LM_ERROR,
+                 "Error in %s: Line %d - Could lock resource\n"
+                 __FILE__,
+                 __LINE__));
       ACE_PRINT_EXCEPTION(ACE_ANY_EXCEPTION,
-                          "Error in locking the node for ServerScheduler\n");
+                          "Exception: PCP_Manager::lock");
     }
   ACE_ENDTRY;
 }
@@ -415,14 +381,18 @@
     }
   ACE_CATCHANY
     {
+      ACE_DEBUG((LM_ERROR,
+                 "Error in %s: Line %d - Could not release lock\n"
+                 __FILE__,
+                 __LINE__));
       ACE_PRINT_EXCEPTION(ACE_ANY_EXCEPTION,
-                         "Error in unlocking the node for ServerScheduler\n");
+                          "Exception: PCP_Manager::release_lock");
     }
   ACE_ENDTRY;
 }
 
 
-PCP_Manager_Factory::PCP_Manager_Factory(char *shared_file)
+PCP_Manager_Factory::PCP_Manager_Factory(const char *shared_file)
 {
   ACE_TRY_NEW_ENV
     {
@@ -431,14 +401,19 @@
 
       /// Get the temporary directory
       if (ACE::get_temp_dir (temp_file,
-                             MAXPATHLEN - ACE_OS::strlen(shared_file))
+                             MAXPATHLEN - ACE_OS_String::strlen(shared_file))
           == -1)
-        ACE_ERROR ((LM_ERROR,
-                    "Temporary path too long\n"));
+        {
+          ACE_DEBUG((LM_ERROR,
+                     "Error in %s: Line %d - Shared File Name too long\n"
+                     __FILE__,
+                     __LINE__));
+          ACE_OS::exit(1);
+        }
       ACE_TRY_CHECK;
 
       /// Add the filename to the end
-      ACE_OS::strcat (temp_file, shared_file);
+      ACE_OS_String::strcat (temp_file, shared_file);
 
       /// Store in the global variable.
       this->shm_key_ = temp_file;
@@ -460,8 +435,11 @@
         ACE_TRY_CHECK;
 
 #else /* !ACE_LACKS_MMAP */
-      ACE_ERROR ((LM_INFO,
-                  "mmap is not supported on this platform\n"));
+        ACE_DEBUG((LM_ERROR,
+                   "Error in %s: Line %d - ACE_LACKS_MMAP - cannot create shared memory\n"
+                   __FILE__,
+                   __LINE__));
+        ACE_OS::exit();
 #endif /* !ACE_LACKS_MMAP */
 
       /// determine space requirements for the lock list
@@ -477,8 +455,11 @@
       if (result == -1)
         {
           ACE_ERROR((LM_ERROR,
-                     "Error in creating the shared memory segment to hold "
-                     "Lock information, aborting ServerScheduler.\n"));
+                   "Error in %s: Line %d - Error in creating the shared "
+                   " memory segment to hold Lock information, "
+                   "aborting ServerScheduler.\n"
+                   __FILE__,
+                   __LINE__));
           ACE_OS::exit(1);
         }
 
@@ -487,19 +468,16 @@
       /// Make the shared memory a place for a lock list
       this->lock_array_ = ACE_static_cast(CosSchedulingLockNode *,
                                           this->mem_.malloc(CosSchedulingLockList_space));
-      if (this->lock_array_ == 0)
-        {
-          ACE_ERROR((LM_ERROR,
-                     "Error in creating Lock Array, locking impossible.\n"));
-        }
-      ACE_TRY_CHECK;
-
       /// get the pointer to the list of locks and
       /// construct a lock list manager object
       if (this->lock_array_ == 0)
         {
           ACE_ERROR((LM_ERROR,
-                     "No Pointer for LockArray, aborting\n"));
+                   "Error in %s: Line %d - Error in creating "
+                   "array to hold lock information "
+                   "ServerScheduler not created\n "
+                   __FILE__,
+                   __LINE__));
           ACE_OS::exit(1);
         }
       else
@@ -516,8 +494,13 @@
     }
   ACE_CATCHANY
     {
+      ACE_ERROR((LM_ERROR,
+                 "Error in %s: Line %d - Error in creating "
+                 "PCP_Manager_Factory to create new PCP_Managers\n"
+                 __FILE__,
+                 __LINE__));
       ACE_PRINT_EXCEPTION(ACE_ANY_EXCEPTION,
-                         "Error in Setting lock factory for ServerScheduler\n");
+                         "PCP_Manager_Factory::PCP_Manager_Factory\n");
     }
   ACE_ENDTRY;
 }
--- ace-5.4.orig/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_PCP_Manager.h
+++ ace-5.4/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_PCP_Manager.h
@@ -4,7 +4,7 @@
 /**
  *  @file   RTCosScheduling_PCP_Manager.h
  *
- *  RTCosScheduling_PCP_Manager.h,v 1.4 2003/12/29 15:16:30 dhinton Exp
+ *  $Id: RTCosScheduling_PCP_Manager.h,v 1.5 2004/02/13 21:18:09 bryank Exp $
  *
  *  @author Matt Murphy <murphym@cs.uri.edu>
  *  @author based upon work by Greg Cooper
@@ -108,7 +108,7 @@
    * Translates the pointer into an offset and
    * stores it in the structure
    */
-  void next(struct CosSchedulingLockNode *Next);
+  void next(const struct CosSchedulingLockNode *Next);
 
   /**
    * Copies the relevant fields while preserving those which should not be
@@ -130,7 +130,9 @@
    * @param size The size of the lock list
    * @param mutex the mutex that guards the CosSchedulingLockList
    */
-  CosSchedulingLockList(CosSchedulingLockNode *lock_array, int size, ACE_SYNCH_MUTEX *mutex);
+  CosSchedulingLockList(CosSchedulingLockNode *lock_array,
+                        const int size,
+                        ACE_SYNCH_MUTEX *mutex);
 
   /**
    * Calls ACE_Thread::remove() on all conditions in the list;
@@ -139,7 +141,7 @@
    *
    * @param size the number of locks to destroy, starting at head
    */
-  void destroy(int size);
+  void destroy(const int size);
 
   /**
    * Adds L to the granted list, if space is available
@@ -229,7 +231,7 @@
    */
   PCP_Manager(CosSchedulingLockList *locks,
               ACE_SYNCH_MUTEX *mutex,
-              RTCORBA::Current_var current);
+              const RTCORBA::Current_var current);
 
   /**
    * Acquires a lock on a shared resource using the
@@ -238,7 +240,7 @@
    * @param PriorityCeiling The priority ceiling of the lock
    * @param priority The priority to lock at.
    */
-  void lock(int PriorityCeiling, int Priority);
+  void lock(const int PriorityCeiling, const int Priority);
 
   /**
    * Releases a lock previously granted with lock()
@@ -275,7 +277,7 @@
    * pointers to the granted and pending lock lists as well as
    * the mutex and condition variable.
    */
-  PCP_Manager_Factory(char *shared_file);
+  PCP_Manager_Factory(const char *shared_file);
 
   ~PCP_Manager_Factory();
 
--- ace-5.4.orig/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_PCP_Manager.i
+++ ace-5.4/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_PCP_Manager.i
@@ -4,7 +4,7 @@
 /**
  *  @file   RTCosScheduling_PCP_Manager.i
  *
- *  RTCosScheduling_PCP_Manager.i,v 1.2 2003/10/15 08:20:45 jwillemsen Exp
+ *  $Id: RTCosScheduling_PCP_Manager.i,v 1.3 2004/02/13 21:18:09 bryank Exp $
  *
  *  @author Matt Murphy <murphym@cs.uri.edu>
  *  @author based upon work by Greg Cooper
@@ -53,3 +53,43 @@
 {
   return TAO::PCP_Manager(this->locks_, &this->mutex_, current);
 }
+
+ACE_INLINE struct TAO::CosSchedulingLockNode *
+TAO::CosSchedulingLockNode::next()
+{
+  /// INT_MAX is a special value indicating the end of a list
+  if (this->next_offset_ == INT_MAX)
+    {
+      return 0;
+    }
+  else
+    {
+      return this + this->next_offset_;
+    }
+}
+
+ACE_INLINE void
+TAO::CosSchedulingLockNode::next(const struct CosSchedulingLockNode *next_lock)
+{
+  /// INT_MAX is a special value indicating the end of a list
+  if (next_lock == 0)
+    {
+      this->next_offset_ = INT_MAX;
+    }
+  else
+    {
+      this->next_offset_ = next_lock - this;
+    }
+}
+
+ACE_INLINE const TAO::CosSchedulingLockNode&
+TAO::CosSchedulingLockNode::operator=(const CosSchedulingLockNode& L)
+{
+  this->threadID_           = L.threadID_;
+  this->priority_ceiling_   = L.priority_ceiling_;
+  this->priority_           = L.priority_;
+  this->elevated_priority_  = L.elevated_priority_;
+
+  return *this;
+}
+
--- ace-5.4.orig/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_ServerScheduler_i.cpp
+++ ace-5.4/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_ServerScheduler_i.cpp
@@ -4,7 +4,7 @@
 /**
  *  @file    RTCosScheduling_ServerScheduler_i.cpp
  *
- *  RTCosScheduling_ServerScheduler_i.cpp,v 1.6 2003/11/01 20:54:57 bala Exp
+ *  $Id: RTCosScheduling_ServerScheduler_i.cpp,v 1.8 2004/02/13 22:06:29 bryank Exp $
  *
  *  @author Matt Murphy <murphym@cs.uri.edu>
  */
@@ -668,7 +668,7 @@
     }
 
 
-  char line[BUF_MAX], key[16];
+  char line[BUF_MAX], key[64];
   ACE_OS::strncpy(key, "Node ", sizeof("Node "));
   ACE_OS::strcat(key, node_name);
 
@@ -676,41 +676,42 @@
   do
     {
       ACE_OS::fgets(line, BUF_MAX, fp);
-    }
-  while (ACE_OS::strncmp(line,key,ACE_OS::strlen(key)) != 0);
-  /// Make sure we did not hit the end of file
   if (ACE_OS::last_error() == EOF)
     {
       ACE_ERROR_RETURN((LM_ERROR,
-                        "Node %s not found in config file, aborting\n",
+                            "Node %s not found in config file\n",
                         node_name),
                         0);
+          break;
     }
 
+    }
+  while (ACE_OS::strncmp(line,key,ACE_OS::strlen(key)) != 0);
 
   /// Skip to the appropriate Task section of the node
   ACE_OS::clearerr(fp);
   do
     {
       ACE_OS::fgets(line, BUF_MAX, fp);
-    }
-  while (ACE_OS::strncmp(line,
-                         "Resources:",
-                         ACE_OS::strlen("Resources:")
-                        ) != 0);
-
   /// Make sure we did not hit the end of file
   if (ACE_OS::last_error() == EOF)
     {
       ACE_ERROR_RETURN((LM_ERROR,
-                        "Resource list not found for node %s, aborting\n",
+                            "Task list not found for node %s\n",
                         node_name),
                         0);
+          break;
+        }
+
     }
+  while (ACE_OS::strncmp(line,
+                         "Resources:",
+                         ACE_OS::strlen("Resources:")
+                        ) != 0);
 
   CORBA::Short done = 0;
   COS_SCHEDULING_RESOURCE_KEY name;
-  COS_SCHEDULING_RESOURCE_VALUE priority;
+  COS_SCHEDULING_RESOURCE_VALUE priority = 0;
   u_int delimiter;
 
   /// read each activity/priority pair from the config file
@@ -724,7 +725,11 @@
         {
           name      = ACE_CString(line);
           delimiter = name.find('\t');
-          priority  = ACE_OS::atoi(ACE_OS::strchr(line, '\t'));
+          char *p = ACE_OS::strchr(line, '\t');
+          if (p)
+              priority  = ACE_OS::atoi(p);
+          if (priority == 0)
+              priority = RTCORBA::minPriority;
           if (delimiter < name.length() && delimiter > 0)
             {
               resource_map->bind(name.substr(0, delimiter), priority);
--- ace-5.4.orig/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_ServerScheduler_i.h
+++ ace-5.4/TAO/orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_ServerScheduler_i.h
@@ -4,7 +4,7 @@
 /**
  *  @file    RTCosScheduling_ServerScheduler_i.h
  *
- *  RTCosScheduling_ServerScheduler_i.h,v 1.6 2003/12/30 13:29:31 jwillemsen Exp
+ *  $Id: RTCosScheduling_ServerScheduler_i.h,v 1.9 2004/02/24 19:27:07 jwillemsen Exp $
  *
  *  @author Matt Murphy <murphym@cs.uri.edu>
  */
@@ -28,8 +28,8 @@
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "ace/SString.h"
-#include <orbsvcs/RTCosSchedulingS.h>
-#include <orbsvcs/RTCosScheduling/RTCosScheduling_export.h>
+#include <orbsvcs/orbsvcs/RTCosSchedulingS.h>
+#include <orbsvcs/orbsvcs/RTCosScheduling/RTCosScheduling_export.h>
 #include "RTCosScheduling_PCP_Manager.h"
 #include "tao/PortableInterceptorC.h"
 #include "tao/LocalObject.h"
--- ace-5.4.orig/examples/Export/run_test.pl
+++ ace-5.4/examples/Export/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/examples/IPC_SAP/SOCK_SAP/summarize
+++ ace-5.4/examples/IPC_SAP/SOCK_SAP/summarize
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/examples/IPC_SAP/SSL_SAP/summarize
+++ ace-5.4/examples/IPC_SAP/SSL_SAP/summarize
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/examples/QOS/Diffserv/run_test.pl
+++ ace-5.4/examples/QOS/Diffserv/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.orig/examples/Reactor/WFMO_Reactor/run_test.pl
+++ ace-5.4/examples/Reactor/WFMO_Reactor/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
