--- ace-5.4.2.1.orig/GNUmakefile
+++ ace-5.4.2.1/GNUmakefile
@@ -26,9 +26,9 @@
 	$(KEEP_GOING)@cd ASNMP && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd ACEXML && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd apps && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd examples && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd examples && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd netsvcs && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd performance-tests && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd tests && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd performance-tests && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd tests && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd websvcs && $(MAKE) -f $(MAKEFILE) $(@)
 
--- ace-5.4.2.1.orig/examples/QOS/Diffserv/run_test.pl.old
+++ ace-5.4.2.1/examples/QOS/Diffserv/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/examples/Export/run_test.pl
+++ ace-5.4.2.1/examples/Export/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/examples/Reactor/WFMO_Reactor/run_test.pl
+++ ace-5.4.2.1/examples/Reactor/WFMO_Reactor/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/examples/IPC_SAP/SSL_SAP/summarize
+++ ace-5.4.2.1/examples/IPC_SAP/SSL_SAP/summarize
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/examples/IPC_SAP/SOCK_SAP/summarize
+++ ace-5.4.2.1/examples/IPC_SAP/SOCK_SAP/summarize
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/CIAO/DAnCE/GNUmakefile
+++ ace-5.4.2.1/TAO/CIAO/DAnCE/GNUmakefile
@@ -29,5 +29,5 @@
 	$(KEEP_GOING)@cd DomainApplicationManager && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd ExecutionManager && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd RepositoryManager && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd tests && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd tests && $(MAKE) -f $(MAKEFILE) $(@)
 
--- ace-5.4.2.1.orig/TAO/CIAO/examples/Hello/descriptors/run_test_remote.pl
+++ ace-5.4.2.1/TAO/CIAO/examples/Hello/descriptors/run_test_remote.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 #run_test_remote.pl,v 1.2 2004/05/25 02:19:36 lut Exp
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
--- ace-5.4.2.1.orig/TAO/CIAO/examples/Hello/descriptors/run_test_default.pl
+++ ace-5.4.2.1/TAO/CIAO/examples/Hello/descriptors/run_test_default.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/GNUmakefile
+++ ace-5.4.2.1/TAO/GNUmakefile
@@ -24,9 +24,9 @@
 	$(KEEP_GOING)@cd TAO_IDL && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd tao && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd orbsvcs && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd examples && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd examples && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd docs && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd performance-tests && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd tests && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd performance-tests && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd tests && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd utils && $(MAKE) -f $(MAKEFILE) $(@)
 
--- ace-5.4.2.1.orig/TAO/orbsvcs/GNUmakefile
+++ ace-5.4.2.1/TAO/orbsvcs/GNUmakefile
@@ -42,7 +42,7 @@
 	$(KEEP_GOING)@cd Dump_Schedule && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd CosEvent_Service && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd Concurrency_Service && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd performance-tests && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd tests && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd examples && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd performance-tests && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd tests && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd examples && $(MAKE) -f $(MAKEFILE) $(@)
 
--- ace-5.4.2.1.orig/TAO/orbsvcs/orbsvcs/Log/NotifyLogFactory_i.cpp
+++ ace-5.4.2.1/TAO/orbsvcs/orbsvcs/Log/NotifyLogFactory_i.cpp
@@ -185,11 +185,12 @@
   ACE_CHECK_RETURN (notify_log._retn ());
 
   // Add to the Hash table..
-  if (hash_map_.bind (id, notify_log.in ()) == -1)
+  if (hash_map_.bind (id, 
+		      DsNotifyLogAdmin::NotifyLog::_duplicate (notify_log.in ())) == -1)
     ACE_THROW_RETURN (CORBA::INTERNAL (),
                       DsNotifyLogAdmin::NotifyLog::_nil ());
 
-  notifier_->object_creation (notify_log.in (),
+  notifier_->object_creation (DsNotifyLogAdmin::NotifyLog::_duplicate (notify_log.in ()),
                               id ACE_ENV_ARG_PARAMETER);
   ACE_CHECK_RETURN (DsNotifyLogAdmin::NotifyLog::_nil ());
 
--- ace-5.4.2.1.orig/TAO/orbsvcs/orbsvcs/Log/LogMgr_i.cpp
+++ ace-5.4.2.1/TAO/orbsvcs/orbsvcs/Log/LogMgr_i.cpp
@@ -5,6 +5,7 @@
            "LogMgr_i.cpp,v 1.17 2003/04/30 12:31:00 elliott_c Exp")
 
 TAO_LogMgr_i::TAO_LogMgr_i (void)
+: max_id_(0)
 {
   // No-Op.
 }
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/RtEC/IIOPGateway/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/RtEC/IIOPGateway/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/RtEC/IIOPGateway/consumerec_crash.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/RtEC/IIOPGateway/consumerec_crash.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/RtEC/Simple/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/RtEC/Simple/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/CosEC/RtEC_Based/tests/Multiple/run_test2.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/CosEC/RtEC_Based/tests/Multiple/run_test2.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/CosEC/RtEC_Based/tests/Multiple/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/CosEC/RtEC_Based/tests/Multiple/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/CosEC/Simple/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/CosEC/Simple/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/Notify/Filter/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/Notify/Filter/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/Notify/Lanes/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/Notify/Lanes/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/Notify/Subscribe/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/Notify/Subscribe/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/Notify/ThreadPool/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/Notify/ThreadPool/run_test.pl
@@ -1,3 +1,4 @@
+#!/usr/bin/perl
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/Security/Send_File/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/Security/Send_File/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/orbsvcs/examples/LoadBalancing/run_test.pl
+++ ace-5.4.2.1/TAO/orbsvcs/examples/LoadBalancing/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/AMH/Sink_Server/run_test.pl
+++ ace-5.4.2.1/TAO/examples/AMH/Sink_Server/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/OBV/Typed_Events/run_test.pl
+++ ace-5.4.2.1/TAO/examples/OBV/Typed_Events/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/DSI/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/DSI/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/TIE/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/TIE/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/Reference_Counted_Servant/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/Reference_Counted_Servant/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/Default_Servant/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/Default_Servant/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/Loader/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/Loader/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/NewPOA/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/NewPOA/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/Adapter_Activator/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/Adapter_Activator/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/POA_BiDir/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/POA_BiDir/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/On_Demand_Loading/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/On_Demand_Loading/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/RootPOA/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/RootPOA/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/On_Demand_Activation/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/On_Demand_Activation/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/FindPOA/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/FindPOA/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/Explicit_Activation/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/Explicit_Activation/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/POA/Forwarding/run_test.pl
+++ ace-5.4.2.1/TAO/examples/POA/Forwarding/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Persistent_Grid/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Persistent_Grid/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Buffered_Oneways/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Buffered_Oneways/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Callback_Quoter/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Callback_Quoter/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/RTCORBA/Activity/run_test.pl
+++ ace-5.4.2.1/TAO/examples/RTCORBA/Activity/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Quoter/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Quoter/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Simple/bank/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Simple/bank/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Simple/echo/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Simple/echo/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Simple/grid/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Simple/grid/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Simple/time/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Simple/time/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Simple/time-date/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Simple/time-date/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Advanced/ch_3/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Advanced/ch_3/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Advanced/ch_12/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Advanced/ch_12/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Advanced/ch_18/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Advanced/ch_18/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Advanced/ch_21/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Advanced/ch_21/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Advanced/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Advanced/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Advanced/ch_8_and_10/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Advanced/ch_8_and_10/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Simulator/DOVEBrowser/ecdata_patch.pl
+++ ace-5.4.2.1/TAO/examples/Simulator/DOVEBrowser/ecdata_patch.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -pi -S $0 ${1+"$@"}'
     & eval 'exec perl -pi -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/ior_corbaloc/run_test.pl
+++ ace-5.4.2.1/TAO/examples/ior_corbaloc/run_test.pl
@@ -1,4 +1,6 @@
-     eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
+#!/usr/bin/perl
+
+eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
   &  eval 'exec perl -S $0 $argv:q'
   if 0;
 
--- ace-5.4.2.1.orig/TAO/examples/RTScheduling/Fixed_Priority_Scheduler/run_test.pl
+++ ace-5.4.2.1/TAO/examples/RTScheduling/Fixed_Priority_Scheduler/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/RTScheduling/MIF_Scheduler/run_test.pl
+++ ace-5.4.2.1/TAO/examples/RTScheduling/MIF_Scheduler/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/PluggableUDP/tests/Basic/run_test.pl
+++ ace-5.4.2.1/TAO/examples/PluggableUDP/tests/Basic/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/PluggableUDP/tests/SimplePerformance/run_test.pl
+++ ace-5.4.2.1/TAO/examples/PluggableUDP/tests/SimplePerformance/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/PluggableUDP/tests/Performance/run_test.pl
+++ ace-5.4.2.1/TAO/examples/PluggableUDP/tests/Performance/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/TypeCode_Creation/run_test.pl
+++ ace-5.4.2.1/TAO/examples/TypeCode_Creation/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Logging/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Logging/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Load_Balancing/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Load_Balancing/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/TAO/examples/Buffered_AMI/run_test.pl
+++ ace-5.4.2.1/TAO/examples/Buffered_AMI/run_test.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/ace/IOStream.h
+++ ace-5.4.2.1/ace/IOStream.h
@@ -31,7 +31,7 @@
 // by its config.h file.
 // This restriction is recorded in Bugzilla entry 857.
 #if defined (ACE_HAS_STANDARD_CPP_LIBRARY) && (ACE_HAS_STANDARD_CPP_LIBRARY == 1)
-#  if !defined (ACE_USES_OLD_IOSTREAMS) && !defined (ACE_LACKS_ACE_IOSTREAM)
+#  if !defined (ACE_USES_OLD_IOSTREAMS) && !defined (ACE_USES_NEW_IOSTREAMS) && !defined (ACE_LACKS_ACE_IOSTREAM)
 #    define ACE_LACKS_ACE_IOSTREAM
 #  endif /* !ACE_USES_OLD_IOSTREAMS && !ACE_LACKS_ACE_IOSTREAM */
 #endif /* ACE_HAS_STANDARD_CPP_LIBRARY */
@@ -477,9 +477,9 @@
         inline virtual MT& operator<<(__omanip_ func) CODE2 \
         inline virtual MT& operator<<(__manip_ func)  CODE2
 
-#  if defined (ACE_LACKS_SIGNED_CHAR)
-  #define GET_FUNC_SET1(MT,CODE,CODE2) GET_FUNC_SET0(MT,CODE,CODE2)
-  #define PUT_FUNC_SET1(MT,CODE,CODE2) PUT_FUNC_SET0(MT,CODE,CODE2)
+#  if defined (ACE_LACKS_SIGNED_CHAR) || defined(ACE_LACKS_CHAR_RIGHT_SHIFTS)
+#    define GET_FUNC_SET1(MT,CODE,CODE2) GET_FUNC_SET0(MT,CODE,CODE2)
+#    define PUT_FUNC_SET1(MT,CODE,CODE2) PUT_FUNC_SET0(MT,CODE,CODE2)
 #  else
 #    if defined (ACE_LACKS_CHAR_STAR_RIGHT_SHIFTS)
   #define GET_FUNC_SET1(MT,CODE,CODE2) \
--- ace-5.4.2.1.orig/ace/config-g++-common.h
+++ ace-5.4.2.1/ace/config-g++-common.h
@@ -56,6 +56,14 @@
 # undef  ACE_USES_OLD_IOSTREAMS
 #endif /* __GNUC__ >= 2.97 */
 
+#if __GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 2 )
+# define ACE_LACKS_UNBUFFERED_STREAMBUF
+# define ACE_LACKS_LINEBUFFERED_STREAMBUF
+# define ACE_LACKS_IOSTREAM_FX
+# define ACE_LACKS_CHAR_RIGHT_SHIFTS
+# define ACE_USES_NEW_IOSTREAMS
+#endif
+
   // __EXCEPTIONS is defined with -fexceptions, the egcs default.  It
   // is not defined with -fno-exceptions, the ACE default for g++.
   // ACE_HAS_EXCEPTIONS is defined in
--- ace-5.4.2.1.orig/bin/MakeProjectCreator/templates/gnu.mpd
+++ ace-5.4.2.1/bin/MakeProjectCreator/templates/gnu.mpd
@@ -106,10 +106,18 @@
 endif
 
 <%if(exename)%>
-INSBIN ?= <%if(install)%><%install%><%else%>.<%endif%>
+<%if(install)%>
+INSBIN = <%install%>
+<%else%>
+INSBIN ?= .
+<%endif%>
 OUTPUT_DIRECTORY = $(INSBIN)
 <%else%>
-INSLIB ?= <%if(dllout)%><%dllout%><%else%><%libout%><%endif%>
+<%if(dllout || libout)%>
+INSLIB = <%if(dllout)%><%dllout%><%else%><%libout%><%endif%>
+<%else%>
+INSLIB ?=
+<%endif%>
 ifeq ($(INSLIB),.)
   ifeq ($(PWD),)
     PWD=$(shell pwd)
--- ace-5.4.2.1.orig/bin/ace_ld
+++ ace-5.4.2.1/bin/ace_ld
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
--- ace-5.4.2.1.orig/bin/mpc.pl
+++ ace-5.4.2.1/bin/mpc.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
@@ -16,7 +18,7 @@
 use Cwd;
 use File::Basename;
 
-my($basePath) = getExecutePath($0) . '/MakeProjectCreator';
+my($basePath) = '/usr/share/ace/bin/MakeProjectCreator';
 unshift(@INC, $basePath . '/modules');
 
 my($mpcroot) = $ENV{MPC_ROOT};
--- ace-5.4.2.1.orig/bin/mwc.pl
+++ ace-5.4.2.1/bin/mwc.pl
@@ -1,3 +1,5 @@
+#!/usr/bin/perl
+#
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
@@ -16,7 +18,7 @@
 use Cwd;
 use File::Basename;
 
-my($basePath) = getExecutePath($0) . '/MakeProjectCreator';
+my($basePath) = '/usr/share/ace/bin/MakeProjectCreator';
 unshift(@INC, $basePath . '/modules');
 
 my($mpcroot) = $ENV{MPC_ROOT};
--- ace-5.4.2.1.orig/apps/GNUmakefile
+++ ace-5.4.2.1/apps/GNUmakefile
@@ -21,11 +21,11 @@
 
 REMAINING_TARGETS := $(TARGETS_NESTED:.nested=)
 $(REMAINING_TARGETS):
-	$(KEEP_GOING)@cd Gateway && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd JAWS && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd JAWS3 && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd drwho && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd Gateway && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd JAWS && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd JAWS3 && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd drwho && $(MAKE) -f $(MAKEFILE) $(@)
 	$(KEEP_GOING)@cd gperf && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd mkcsregdb && $(MAKE) -f $(MAKEFILE) $(@)
-	$(KEEP_GOING)@cd soreduce && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd mkcsregdb && $(MAKE) -f $(MAKEFILE) $(@)
+#	$(KEEP_GOING)@cd soreduce && $(MAKE) -f $(MAKEFILE) $(@)
 
