diff -rupd ACE_wrappers.orig/include/makeinclude/platform_linux_common.GNU ACE_wrappers/include/makeinclude/platform_linux_common.GNU
--- ACE_wrappers.orig/include/makeinclude/platform_linux_common.GNU	2011-03-02 13:45:06.000000000 +0100
+++ ACE_wrappers/include/makeinclude/platform_linux_common.GNU	2014-07-11 14:44:31.447726979 +0200
@@ -108,6 +108,12 @@ ifeq ($(ssl),1)
   PLATFORM_SSL_CPPFLAGS += -I/usr/kerberos/include
 endif # ssl
 
+PLATFORM_LARGEFILE_CFLAGS := $(shell getconf LFS_CFLAGS 2> /dev/null || echo Unknown)
+ifdef PLATFORM_LARGEFILE_CFLAGS
+  # Large file support
+  CPPFLAGS += $(PLATFORM_LARGEFILE_CFLAGS)
+endif #largefile
+
 SYSARCH := $(shell uname -m)
 
 PIC      = -fPIC
